pyvista/plotting/plotting.py

```diff
@@ -4652,7 +4662,18 @@ def remove_and_reactivate_prior_scalars(dsattr, name):
         for mesh, (name, assoc) in self._added_scalars:
             dsattr = mesh.point_data if assoc == 'point' else mesh.cell_data
             remove_and_reactivate_prior_scalars(dsattr, name)
+
+        # reactivate prior active scalars
+        for mesh, point_name, cell_name in self._prev_active_scalars:
```

Comment:
Good point. I completely removed the first section and replaced it with the simple `pop` to remove the offending array.

1. They do in 6ce17e5590254f7e030298d5bcb4c42cd0e46ca4.
2. Added test for this in 6ce17e5590254f7e030298d5bcb4c42cd0e46ca4.
3. They might, though many use `add_mesh` under the hood. We can raise an issue, but I think it would be out of scope to add it in this PR.